scrabble.js

```diff
@@ -1,16 +1,228 @@
+// Start Scores Object
+const scores = {
+  'A': 1,
+  'E': 1,
+  'I': 1,
+  'O': 1,
+  'U': 1,
+  'L': 1,
+  'N': 1,
+  'R': 1,
+  'S': 1,
+  'T': 1,
+  'D': 2,
+  'G': 2,
+  'B': 3,
+  'C': 3,
+  'M': 3,
+  'P': 3,
+  'F': 4,
+  'H': 4,
+  'V': 4,
+  'W': 4,
+  'Y': 4,
+  'K': 5,
+  'J': 8,
+  'X': 8,
+  'Q': 10,
+  'Z': 10
+}
 
+const letters = [
+  'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A', 'A',
+  'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E', 'E',
+  'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I', 'I',
+  'O', 'O', 'O', 'O', 'O', 'O', 'O', 'O',
+  'U', 'U', 'U', 'U',
+  'L', 'L', 'L', 'L',
+  'N', 'N', 'N', 'N', 'N', 'N',
+  'R', 'R', 'R', 'R', 'R', 'R',
+  'S', 'S', 'S', 'S',
+  'T', 'T', 'T', 'T', 'T', 'T',
+  'D', 'D', 'D', 'D',
+  'G', 'G', 'G',
+  'B', 'B',
+  'C', 'C',
+  'M', 'M',
+  'P', 'P',
+  'F', 'F',
+  'H', 'H',
+  'V', 'V',
+  'W', 'W',
+  'Y', 'Y',
+  'K',
+  'J',
+  'X',
+  'Q',
+  'Z'
+];
+
+// Start Scrabble Object
 const Scrabble = {
+  // helper method will return true if input word is valid
+  isValid(word) {
+    if (typeof word !== 'string') {
+      return false;
+    }
+
+    if ( /^[a-zA-Z]+$/.test(word) ) {
+      return true;
+    }
+  },
+
   score(word) {
+    // validations on word input
+    if ( !Scrabble.isValid(word) ) {
+      throw new Error('Please enter valid letters.');
+    }
+
+    if (word.length > 7 || word.length < 1) {
+      throw new Error('Word must be no more than 7 letters.');
+    }
+
+    // begin scoring word functionality
+    word = word.toUpperCase();
+    let wordScore = [];
+
+    for (let i = 0; i < word.length; i += 1) {
+      wordScore.push(scores[word[i]]);
+    }
+
+    let totalScore = wordScore.reduce(function(acc, val) { return acc + val; });
 
+    if (word.length == 7) {
+      totalScore += 50;
+    }
+
+    return totalScore;
   },
+
   highestScoreFrom(arrayOfWords) {
+    // validation on array of words
+    if (arrayOfWords.length === 0) {
+      throw new Error('No words were passed.');
+    }
+
+    // begin highest score functionality
+    let topScores = [];
+    let maxScore = 0;
+
+    for (let word of arrayOfWords) {
+      let wordScore = this.score(word);
+      if (wordScore == maxScore) {
```

Comment:
This logic ends up getting pretty complex. Is there a way you could clarify what's going on, maybe through a helper function?